File: TikTokClone.Domain/Entities/RefreshToken.cs
namespace TikTokClone.Domain.Entities
{
    public class RefreshToken
    {
        public int Id { get; set; }
        public string Token { get; set; } = string.Empty;
        public string UserId { get; set; } = string.Empty;
        public DateTime ExpiresAt { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? RevokedAt { get; set; }
        public string? ReplacedByToken { get; set; }
        public string? RevokedByIp { get; set; }
        public string? CreatedByIp { get; set; }

        // Navigation property
        public virtual User User { get; set; } = null!;

        // Computed properties
        public bool IsExpired => DateTime.UtcNow >= ExpiresAt;
        public bool IsRevoked => RevokedAt.HasValue;
        public bool IsActive => !IsRevoked && !IsExpired;

        // Domain methods
        public void Revoke(string? replacedByToken = null, string? revokedByIp = null)
        {
            RevokedAt = DateTime.UtcNow;
            ReplacedByToken = replacedByToken;
            RevokedByIp = revokedByIp;
        }

        public bool CanBeRefreshed()
        {
            return IsActive;
        }
    }
}

File: backend/auth-service-golang/internal/domain/entities/refresh_token.go
package entities

import (
	"time"

	"github.com/google/uuid"
)

type RefreshToken struct {
	ID        uuid.UUID `gorm:"primaryKey"`
	UserID    uuid.UUID `gorm:"type:uuid;not null;index"`
	Token     string    `gorm:"not null;index"`
	ExpiresAt time.Time `gorm:"not null"`
	IsRevoked bool      `gorm:"not null"`
	CreatedAt time.Time `gorm:"autoCreateTime"`
	UpdatedAt time.Time `gorm:"autoUpdateTime"`
}

func (t *RefreshToken) IsExpired() bool {
	return time.Now().After(t.ExpiresAt)
}

func (t *RefreshToken) Revoke() {
	t.IsRevoked = true
}
